usm_rest_api/v1/views/views.py

```diff
@@ -0,0 +1,371 @@
+    for node in nodelist[:]:
+        try:
+            ssh_fingerprint = saltapi.get_fingerprint(
+                saltapi.get_host_ssh_key(node['management_ip']))
+            rc,out,err = saltapi.setup_minion(node['management_ip'],
+                                              ssh_fingerprint, node['ssh_username'],
+                                              node['ssh_password'], postdata)
+            if rc!=0:
+                raise Exception("Accept Minion Failed", rc,out,err)
+        except Exception, e:
+            log.exception(e)
+            #Delete the node from list to avoid further processing
+            nodelist.remove(node)
+            failedNodes.append(node)
+    #Sleep for sometime to make sure all the restarted minions are back
+    #online
+    log.debug("Accepting the minions keys" )
+    createGlusterCluster.update_state(state='ADD_MINION_KEYS_AND_DB_UPDATE')
+    time.sleep( 3 )
+    #Accept the keys of the successful minions and add to the DB
+    for node in nodelist[:]:
+        #
+        #TODO Get the host uuid from  host and update in DB
+        #
+        node['node_id'] = uuid.uuid4()
+        node['cluster'] = str(postdata['cluster_id'])
+        try:
+            saltapi.accept_minion(socket.gethostbyaddr(node['management_ip'])[0])
+        except Exception, e:
+            log.exception(e)
+            #Delete the node from list to avoid peer probing
+            nodelist.remove(node)
+            failedNodes.append(node)
+            continue
+        hostSerilaizer = HostSerializer(data=node)   
+        try:
+            if hostSerilaizer.is_valid():
+                #Delete all the fields those are not reqired to be persisted
+                del hostSerilaizer.validated_data['ssh_password']
+                del hostSerilaizer.validated_data['ssh_key_fingerprint']
+                del hostSerilaizer.validated_data['ssh_username']
+                del hostSerilaizer.validated_data['ssh_port']
+                
+                hostSerilaizer.save()
+            else:
+               log.error("Host Creation failed. Invalid hostSerilaizer")
+               log.error("hostSerilaizer Err: %s" % hostSerilaizer.errors)
+               raise Exception("Add to DB Failed", hostSerilaizer.errors)
+        except Exception, e:
+            log.exception(e)
+            if node not in failedNodes:
+                failedNodes.append(node)
+            if node in nodelist:
+                nodelist.remove(node)
+            continue
+        
+    #Now create the cluster
+    log.debug("peer probe start" )
+    createGlusterCluster.update_state(state='PEER_PROBE')
+    #
+    #Wait for some time so that the communication chanel is ready
+    #
+    time.sleep(10)
+    #Need peerprobe only if the length is more than one
+    if len(nodelist)>1:
+        rootNode = nodelist[0]['management_ip']
+        del nodelist[0]
+        for node in nodelist:
```

Comment:
I am taking out first node from the list as root node and form the cluster with rest of the nodes in the cluster